docs/public/specifications/tests/Address-TP/address01.md

```diff
@@ -1,47 +1,147 @@
 
-The domain name to be tested.
 
-### Ordered description of steps to be taken to execute the test case
+The value in the Level column is the default severity level of the message. The
+severity level can be changed in the [Zonemaster-Engine profile]. Also see the
+[Severity Level Definitions] document.
 
-1. Obtain the IP addresses of each name server of the domain from the parent using
-   [Method4](../Methods.md) and child using [Method5](../Methods.md)
+The argument names in the Arguments column lists the arguments used in the
+message. The argument names are defined in the [Argument list].
 
-2. If any IP address (IPv4 or IPv6) falls within any of the blocks in the below
-   mentioned IANA links, the test case fails:
-   * <https://www.iana.org/assignments/iana-ipv4-special-registry/iana-ipv4-special-registry.xhtml>
-   * <https://www.iana.org/assignments/iana-ipv6-special-registry/iana-ipv6-special-registry.xhtml>
+## Test procedure 
 
-### Outcome(s)
+1. Create the empty set: Name server name and IP address ("Name Server IP").
 
-If one name server has one of its addresses matches a forbidden address
-block, the test fails. If all the name server addresses are outside these
-forbidden blocks, the test case succeeds. 
+2. Obtain the glue address records of each name server for the domain from the
+   parent using the method [Get-Del-NS-Names-and-IPs] and add them to the 
+   *Name Server IP* set. 
 
-### Special procedural requirements
+3. Obtain the IP addresses of each name server for the domain using the method 
+   [Get-Zone-NS-Names-and-IPs] and add any non-duplicate results to 
+   *Name Server IP* set. 
 
-The registries listed in bullet 2 above must be fetched prior to testing 
+4. Create the following empty sets:
+   1. Name server name and IP address ("Documentation Address").
+   2. Name server name and IP address ("Local Use Address").
+   3. Name server name and IP address ("Not Globally Reachable").
 
-### Intercase dependencies
+5. For each name server in *Name Server IP* do:
+   1. Match the IP address against the IP ranges specified in 
+      [Special purpose IPv4 addresses] and [Special purpose IPv6 addresses]
+      1. If if falls within any of the address ranges reserved for 
+        *Documentation*, add the name server name and IP address to the
+        *Documentation Address* set,
+      2. Else, if it falls within an address range belonging to any of the 
+         following categories:  
+         - *Private-Use (IPv4)*
+         - *Loopback (IPv4)*
+         - *Loopback Address (IPv6)*
+         - *Link Local (IPv4)*
+         - *Link-Local Unicast* (IPv6)
+         - *Unique-Local* (IPv6)
+         - *Shared Address Space* (IPv6)
+         add the name server name and IP address to the *Local Use Adddress* 
+         set. 
+      3. Else, if the IP falls within a range that is not registered as 
```

Comment:
To make even clearer and easier to understand maybe change "within a range" to "within any other range".